### PR TITLE
fixed sysdump crash bug

### LINKED ISSUE #88: `sysdump` crashes if no KubeArmor running
Running sysdump utility with no KubeArmor running in k8s causes a runtime error.
image

**Possible Solution**
We probably need to validate if there are any items in context of KubeArmor before we use it
https://github.com/kubearmor/kubearmor-client/blob/f90ced4f3605d9b2f03a13feea94911eaffcf995/sysdump/sysdump.go#L226

**Hi, If it is the first time that you contribute to KubeArmor, follow these steps:**
Write a comment in this issue thread to let other possible contributors know that you are working on this bug. For eg : `Hey all, I would like to work on this issue.`, checkout [Contributing Guide :fire::sparkles: ](https://github.com/kubearmor/KubeArmor/blob/main/CONTRIBUTING.md)and feel free to ask anything related to this issue in this thread or on our [slack channel](https://kubearmor.herokuapp.com/) ✌🏽

### PR DESCRIPTION
**Pull request by AvineshTripathi**
fixes #88

### CODE REVIEW

File: sysdump/sysdump.go [hunk 1]
```diff
@@ -270,3 +274,22 @@ func IsDirEmpty(name string) (bool, error) {
 
 	return true, nil
 }
+
+// Check the status of KubeArmor
+func CheckStatus(c *k8s.Client) (bool, error) {
+	var status bool
+	pods, err := c.K8sClientset.CoreV1().Pods("kube-system").List(context.Background(), metav1.ListOptions{
```
Comment (daemon1024) [inline sysdump/sysdump.go:281]:
```suggestion
	pods, err := c.K8sClientset.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
```

Let's check in all namespaces.

File: sysdump/sysdump.go [hunk 2]
```diff
@@ -270,3 +274,22 @@ func IsDirEmpty(name string) (bool, error) {
...
+		LabelSelector: "kubearmor-app=kubearmor",
+	})
+	if err != nil {
+		return false, err
+	}
+	if len(pods.Items) != 0 && pods.Items[0].Status.Phase == "Running" {
+		status = true
+	}
+	if !status {
+		fmt.Println("Kube-armor is not running in your cluster")
```
Comment (daemon1024) [inline sysdump/sysdump.go:291]:
```suggestion
		fmt.Println("KubeArmor is not running in your cluster")
```

**Review by daemon1024 — CHANGES_REQUESTED** [review]
Thanks a lot for the PR 🚀 

Can you share a sample `sysdump` if KubeArmor is not running?

Some comments inline, We used to use "kube-system" but KubeArmor can run in other namespaces as well, let's make the function Namespace agnostic if possible.

Also, there are some linter warning, Can you check if they can be handle!

Thanks again 😁

**Comment by AvineshTripathi** [comment]
Hey! i think with the new merge of #102 when I do `make install` I see following error which should be resolved 


Also if i remove the PR then following output is shown when kubearmor is not running

**Comment by nyrahul** [comment]
> Hey! i think with the new merge of #102 when I do `make install` I see following error which should be resolved 
> 
> Also if i remove the PR then following output is shown when kubearmor is not running 

Hey @AvineshTripathi , I do not face this issue when `make install` is attempted. Do you want to get on kubearmor slack and check?

**Comment by AvineshTripathi** [comment]
@nyrahul with the latest merge the current PRs commit is throwing error

**Comment by daemon1024** [comment]
I am guessing something went wrong resolving merge conflicts? Yup let's take it to slack #development if you are still facing issues,

**Comment by daemon1024** [comment]
Hey @AvineshTripathi, Were you able to sort out the build failures. Anything I can help out with?

**Comment by nyrahul** [comment]
Hey @AvineshTripathi , it would be great to get this handled. Please ping on slack if any discussion needed. Please fix the conflicts, rebase and mostly we should be good to go.

**Comment by AvineshTripathi** [comment]
Hey sorrry for delaying with this PR will be solving the conflicts in some hours

**Comment by AvineshTripathi** [comment]
Closing this PR and creating a fresh PR with resolved conflicts

**Pull request closed without merging**